library/formdata.inc.php

```diff
@@ -242,9 +245,19 @@ function escape_identifier($s, $whitelist_items, $die_if_no_match = false, $case
 
         return $ok[$key];
     } else {
-        // Return an item that has been "cleaned" up
-        // (this is currently experimental and goal is to avoid using this)
-        return preg_replace('/[^a-zA-Z0-9_.]/', '', $s);
+        if ($die_if_no_match) {
+            if (preg_match('/[^' . $whitelist_items . ']/', $s)) {
+                // Contains illegal character and $die_if_no_match is set, so die() and send error messages to screen and log
+                error_Log("ERROR: OpenEMR SQL Escaping ERROR of the following string: " . errorLogEscape($s), 0);
+                die("<br><span style='color:red;font-weight:bold;'>".xlt("There was an OpenEMR SQL Escaping ERROR of the following string")." ".text($s)."</span><br>");
+            } else {
+                // Contains all legal characters, so return the legal string
+                return $s;
+            }
+        } else {
+            // Since not using $die_if_no_match, then will remove the illegal characters and send back a legal string
+            return preg_replace('/[^' . $whitelist_items . ']/', '', $s);
+        }
```

Comment:
@sjpadgett ,
I recall you adding in this mechanism to this in the past, but I couldn't find where it was being used in the current codebase. Let me know if I missed something. I made the mechanism more flexible (rather than hardcoding the legal values in the regex, they are now passed as a parameter). I did this in this case to allow escaping of sql column names that did not yet exist (when the layouts engine adds a new sql column.

Reply:
doesn't look like me. I'm too lazy to type long variable names:)